compiler/src/yul/mappers/functions.rs

```diff
@@ -219,3 +219,27 @@ fn func_return(
 
     unreachable!()
 }
+
+fn while_loop(
+    context: &Context,
+    stmt: &Spanned<fe::FuncStmt>,
+) -> Result<yul::Statement, CompileError> {
+    if let fe::FuncStmt::While {
+        test,
+        body,
+        or_else: _,
+    } = &stmt.node
+    {
+        let test = expressions::expr(context, test)?;
+        let yul_body = multiple_func_stmt(context, body)?;
+
+        return Ok(yul::Statement::Block(block! {
```

Comment:
I added a `block_statement!` macro to yultsur recently for situations like this

Reply:
Ah, nice! I updated the PR to use that.